extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -6,101 +6,86 @@
 import * as path from 'path';
-import * as fs from 'fs';
-import * as yaml from 'js-yaml';
-import * as glob from 'fast-glob';
-import { BookTreeItem, BookTreeItemType } from './bookTreeItem';
-import { maxBookSearchDepth, notebookConfigKey } from '../common/constants';
-import { isEditorTitleFree } from '../common/utils';
+import * as fs from 'fs-extra';
+import { IPrompter, QuestionTypes, IQuestion } from '../prompts/question';
+import CodeAdapter from '../prompts/adapter';
+import { BookTreeItem } from './bookTreeItem';
 import * as nls from 'vscode-nls';
+import { isEditorTitleFree } from '../common/utils';
 import { promisify } from 'util';
-import { IJupyterBookToc, IJupyterBookSection } from '../contracts/content';
+import { BookModel } from './bookModel';
 
 const localize = nls.loadMessageBundle();
-const existsAsync = promisify(fs.exists);
+const exists = promisify(fs.exists);
 
-export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeItem>, azdata.nb.NavigationProvider {
-	readonly providerId: string = 'BookNavigator';
+export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeItem> {
 
 	private _onDidChangeTreeData: vscode.EventEmitter<BookTreeItem | undefined> = new vscode.EventEmitter<BookTreeItem | undefined>();
 	readonly onDidChangeTreeData: vscode.Event<BookTreeItem | undefined> = this._onDidChangeTreeData.event;
-	private _tableOfContentPaths: string[] = [];
-	private _allNotebooks = new Map<string, BookTreeItem>();
-	private _extensionContext: vscode.ExtensionContext;
 	private _throttleTimer: any;
 	private _resource: string;
+	private _extensionContext: vscode.ExtensionContext;
+	private prompter: IPrompter;
 	// For testing
 	private _errorMessage: string;
 	private _onReadAllTOCFiles: vscode.EventEmitter<void> = new vscode.EventEmitter<void>();
 	private _openAsUntitled: boolean;
+	public viewId: string;
+	public books: BookModel[];
+	public currentBook: BookModel;
+
+	constructor(workspaceFolders: vscode.WorkspaceFolder[], extensionContext: vscode.ExtensionContext, openAsUntitled: boolean, view: string) {
+		this._openAsUntitled = openAsUntitled;
+		this._extensionContext = extensionContext;
+		this.books = [];
+		this.initialize(workspaceFolders, null);
+		vscode.commands.executeCommand('setContext', 'untitledBooks', openAsUntitled);
+		this.viewId = view;
+		this.prompter = new CodeAdapter();
 
-	constructor(workspaceFolders: vscode.WorkspaceFolder[], extensionContext: vscode.ExtensionContext) {
-		this.initialize(workspaceFolders, null, extensionContext);
 	}
 
-	private initialize(workspaceFolders: vscode.WorkspaceFolder[], bookPath: string, context: vscode.ExtensionContext): void {
-		let workspacePaths: string[] = [];
+	private async initialize(workspaceFolders: vscode.WorkspaceFolder[], bookPath: string): Promise<void> {
 		if (bookPath) {
-			workspacePaths.push(bookPath);
+			let book: BookModel = new BookModel(bookPath, this._openAsUntitled, this._extensionContext);
```

Comment:
This whole method could be cleaned up to only take in a string[] - that way you only need one code path (they're basically doing the same thing anyways). Then just have the caller that passes in the workspacefolders do the mapping on that side.